vhost-device-vsock/src/vhu_vsock_thread.rs

```diff
@@ -313,66 +381,48 @@ impl VhostUserVsockThread {
                     // Has to be EPOLLIN as it was not connected previously
                     return;
                 }
-                let mut unix_stream = match self.thread_backend.stream_map.remove(&fd) {
-                    Some(uds) => uds,
+                let mut stream = match self.thread_backend.stream_map.remove(&fd) {
+                    Some(s) => s,
                     None => {
                         warn!("Error while searching fd in the stream map");
                         return;
                     }
                 };
 
-                // Local peer is sending a "connect PORT\n" command
-                let peer_port = match Self::read_local_stream_port(&mut unix_stream) {
-                    Ok(port) => port,
-                    Err(err) => {
-                        warn!("Error while parsing \"connect PORT\n\" command: {:?}", err);
-                        return;
+                match stream {
+                    StreamType::Vsock(_) => {
+                        panic!("Stream type should not be of type vsock");
```

Comment:
I'd avoid `panic!`

Reply:
The reason I used panic! here is that this code should not be reached as this block of code should only be invoked for a StreamType::Unix . Should I instead do a warn and return here?

Reply:
I have now put a warn instead of panic but this codepath shouldn't really ever be executed as the code will only be invoked for StreamType::Unix. Let me know if you have any suggestion on this.

Reply:
Maybe err instead of warn, but I'd really like to avoid panicking a device since we don't have a good way to restart it.

Reply:
Yeah, error makes sense. Used error now in the latest branch state.